api/ccapi/include/layer.h

```diff
@@ -53,27 +53,27 @@ enum LayerType {
   LAYER_BACKBONE_NNSTREAMER =
     ML_TRAIN_LAYER_TYPE_BACKBONE_NNSTREAMER, /**< Backbone using NNStreamer */
   LAYER_CENTROID_KNN =
-    ML_TRAIN_LAYER_TYPE_CENTROID_KNN, /**< Centroid KNN Layer */
-  LAYER_TIME_DIST,                    /**< Time Distributed Layer type */
+    ML_TRAIN_LAYER_TYPE_CENTROID_KNN,        /**< Centroid KNN Layer */
+  LAYER_CONV1D = ML_TRAIN_LAYER_TYPE_CONV1D, /**< Convolution 1D Layer type */
   LAYER_PREPROCESS_FLIP =
     ML_TRAIN_LAYER_TYPE_PREPROCESS_FLIP, /**< Preprocess flip Layer type */
   LAYER_PREPROCESS_TRANSLATE =
     ML_TRAIN_LAYER_TYPE_PREPROCESS_TRANSLATE, /**< Preprocess translate Layer
                                                  type */
   LAYER_PREPROCESS_L2NORM =
-    ML_TRAIN_LAYER_TYPE_PREPROCESS_L2NORM, /**< Preprocess l2norm Layer
-                                                 type */
-  LAYER_BACKBONE_TFLITE,                   /**< Backbone using TFLite */
-  LAYER_ATTENTION,                         /**< Attention Layer type */
-  LAYER_MOL_ATTENTION,                     /**< MoL Attention Layer type */
-  LAYER_CONV1D,                            /**< Convolution 1D Layer type */
-  LAYER_RESHAPE,                           /**< Reshape Layer type */
-  LAYER_RNNCELL,                           /**< RNN Cell Layer type */
-  LAYER_LSTMCELL,                          /**< LSTM Cell Layer type */
-  LAYER_ZONEOUT_LSTMCELL,                  /**< Zoneout LSTM Cell Layer type */
-  LAYER_GRUCELL,                           /**< GRU Cell Layer type */
-  LAYER_REDUCE_MEAN,                       /**< Reduce mean Layer type */
-  LAYER_IDENTITY,                          /**< Identity Layer type */
+    ML_TRAIN_LAYER_TYPE_PREPROCESS_L2NORM, /**< Preprocess l2norm Layer type */
+  LAYER_LSTMCELL = ML_TRAIN_LAYER_TYPE_LSTMCELL, /**< LSTM Cell Layer type */
+  LAYER_GRUCELL = ML_TRAIN_LAYER_TYPE_GRUCELL,   /**< GRU Cell Layer type */
+  LAYER_RNNCELL = ML_TRAIN_LAYER_TYPE_RNNCELL,   /**< RNN Cell Layer type */
+  LAYER_ZONEOUT_LSTMCELL =
+    ML_TRAIN_LAYER_TYPE_ZONEOUTLSTMCELL, /**< Zoneout LSTM Cell Layer type */
+  LAYER_TIME_DIST,                       /**< Time Distributed Layer type */
+  LAYER_BACKBONE_TFLITE,                 /**< Backbone using TFLite */
+  LAYER_ATTENTION,                       /**< Attention Layer type */
+  LAYER_MOL_ATTENTION,                   /**< MoL Attention Layer type */
+  LAYER_RESHAPE,                         /**< Reshape Layer type */
+  LAYER_REDUCE_MEAN,                     /**< Reduce mean Layer type */
+  LAYER_IDENTITY,                        /**< Identity Layer type */
```

Comment:
LOSS_MSE is 500 and need to check the ML_TRAIN_LAYER_TYPE_RNNCELL is 600. Need to fix

Reply:
Okay, thank you.
but ML_TRAIN_LAYER_TYPE_RNNCELL is 601, it need to change from 601 to 600 ?
All of below enum value need to be reduce 1. right ?
[nntrainer-api-common.h]
  ML_TRAIN_LAYER_TYPE_LSTMCELL = 601, /**< LSTM Cell Layer type (Since 7.0) */
  ML_TRAIN_LAYER_TYPE_GRUCELL = 602,  /**< GRU Cell Layer type (Since 7.0) */
  ML_TRAIN_LAYER_TYPE_RNNCELL = 603,  /**< RNN Cell Layer type (Since 7.0) */
  ML_TRAIN_LAYER_TYPE_ZONEOUTLSTMCELL =
    604, /**< ZoneoutLSTM Cell Layer type (Since 7.0) */

Reply:
How about do this way.
0~299 : Layer for neural network
300 - 399 : simple transformation layer
500 - 599 : Loss layer
So.. let's fix nntrainer-api-common.h too.
- ML_TRAIN_LAYER_TYPE_LSTMCELL = 20,
- ML_TRAIN_LAYER_TYPE_GRUCELL=21,
- ML_TRAIN_LAYER_TYPE_RNNCELL=22,
- ML_TRAIN_LAYER_TYPE_ZONEOUTLSTMCELL =23
and place them in order here (layer.h)

Reply:
thank you